examples/bpi-monolith-test-suite/bpi/bpi.ts

```diff
@@ -0,0 +1,136 @@
+        initalAgreement.productIds = productIds;
+        this.storageComponent.setInitialAgreementState(initalAgreement);
+
+        this.identityComponent = identityComponent;
+        const ownerOrg = this.identityComponent.addOrganization(id, name);
+        this.identityComponent.setOwnerOrganization(ownerOrg);
+
+        this.messagingComponent = messagingComponent;
+        this.transactionPoolComponent = new TransactionPoolComponent(identityComponent);
+        this.workgroupComponent = workgroupComponent;
+        this.vsmComponent = new VirtualStateMachine(this.storageComponent, this.workgroupComponent)
+    }
+
+    // IDENTITY API
+
+    addOrganization(id: string, name: string): BpiSubject {
+        return this.identityComponent.addOrganization(id, name);
+    }
+
+    getOrganizationById(id: string): BpiSubject {
+        return this.identityComponent.getOrganizationById(id);
+    }
+
+    // IDENTITY API END
+
+    // WORKGROUP API
+
+    getWorkgroups(): Workgroup[] {
+        return this.workgroupComponent.getWorkgroups();
+    }
+
+    addWorkgroup(name: string, id: string, worksteps: Workstep[]): Workgroup {
+        return this.workgroupComponent.createWorkgroup(name, id, this.identityComponent.getOwnerOrganization(), worksteps);
+    }
+
+    getWorkgroupById(id: string): Workgroup {
+        return this.workgroupComponent.getWorkgroupById(id);
+    }
+
+    inviteToWorkgroup(id: string, name: string, sender: BpiSubject, recipient: string, workgroupId: string, agreement: Agreement): Invitation {
+        return this.workgroupComponent.sendInviteToWorkgroup(id, name, sender, recipient, workgroupId, agreement);
+    }
+
+    getReceivedInvitationsByEmail(email: string): Invitation[] {
+        return this.workgroupComponent.getReceivedInvitationsByEmail(email);
+    }
+
+    getInvitationById(id: string): Invitation {
+        return this.workgroupComponent.getInvitationById(id);
+    }
+
+    // TODO: Should go through the VSM execution process, probably with a special INIT transaction and workstep
+    signInvitation(invitationId: string, recipientSignature: string, recipientOrgId: string, recipientOrgName: string) {
+        const invitation = this.workgroupComponent.getInvitationById(invitationId);
+
+        // TODO: Verify signature from the recipient
+        // TODO: What is the state change on agreement acceptance - status?
+        const latestAgreementState = this.storageComponent.getAgreementState();
+        const bobsProof = this.vsmComponent.createProof(latestAgreementState, latestAgreementState, recipientSignature);
+
+        const accepteeOrg = this.identityComponent.addOrganization(recipientOrgId, recipientOrgName);
+
+        const workgroup = this.workgroupComponent.getWorkgroupById(invitation.workgroupId);
+        workgroup.addParticipants(accepteeOrg);
+
+        latestAgreementState.proofs.push(bobsProof);
+    }
+
+    // WORKGROUP API END
+
+    verifyProof(proof: string): boolean {
```

Comment:
this is a stub for a future actual call to verify the proof, right? @ognjenkurtic

Reply:
Yes